src/fontra/views/editor/panel-transformation.js

```diff
@@ -450,11 +572,295 @@ export default class TransformationPanel extends Panel {
+  }
+}
+
+// Define moveDescriptor objects
+const alignLeft = {
+  undoLabel: "align left",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const xMins = boundingBoxes.map((bounds) => bounds.xMin);
+    const left = Math.min(...xMins);
+    return xMins.map((xMin) => ({
+      x: left - xMin,
+      y: 0,
+    }));
+  },
+};
+
+const alignCenter = {
+  undoLabel: "align center",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const xMaxes = boundingBoxes.map((bounds) => bounds.xMax);
+    const xMins = boundingBoxes.map((bounds) => bounds.xMin);
+    const left = Math.min(...xMins);
+    const right = Math.max(...xMaxes);
+    return boundingBoxes.map((bounds) => ({
+      x: left - bounds.xMin + (right - left) / 2 - (bounds.xMax - bounds.xMin) / 2,
+      y: 0,
+    }));
+  },
+};
+
+const alignRight = {
+  undoLabel: "align right",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const xMaxes = boundingBoxes.map((bounds) => bounds.xMax);
+    const right = Math.max(...xMaxes);
+    return xMaxes.map((xMax) => ({
+      x: right - xMax,
+      y: 0,
+    }));
+  },
+};
+
+const alignTop = {
+  undoLabel: "align top",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const yMaxes = boundingBoxes.map((bounds) => bounds.yMax);
+    const top = Math.max(...yMaxes);
+    return yMaxes.map((yMax) => ({
+      x: 0,
+      y: top - yMax,
+    }));
+  },
+};
+
+const alignMiddle = {
+  undoLabel: "align middle",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const yMaxes = boundingBoxes.map((bounds) => bounds.yMax);
+    const yMins = boundingBoxes.map((bounds) => bounds.yMin);
+    const bottom = Math.min(...yMins);
+    const top = Math.max(...yMaxes);
+    return boundingBoxes.map((bounds) => ({
+      x: 0,
+      y: top - bounds.yMax + (bounds.yMax - bounds.yMin) / 2 - (top - bottom) / 2,
+    }));
+  },
+};
+
+const alignBottom = {
+  undoLabel: "align bottom",
+  computeDeltasFromBoundingBoxes: (boundingBoxes) => {
+    const yMins = boundingBoxes.map((bounds) => bounds.yMin);
+    const bottom = Math.min(...yMins);
+    return yMins.map((yMin) => ({
+      x: 0,
+      y: bottom - yMin,
+    }));
+  },
+};
+
+class DistributeObjectsDescriptor {
+  constructor(direction, directionVar) {
+    this.undoLabel = `distribute ${direction}`;
+    this.minProperty = `${directionVar}Min`;
+    this.maxProperty = `${directionVar}Max`;
+    this.deltaProperty = directionVar;
+  }
+
+  computeDeltasFromBoundingBoxes(boundingBoxes, distributeValue) {
+    let effectiveExtent = 0;
+    for (const bounds of boundingBoxes) {
+      effectiveExtent += bounds[this.maxProperty] - bounds[this.minProperty];
+    }
+    const mins = boundingBoxes.map((bounds) => bounds[this.minProperty]);
+    const maxes = boundingBoxes.map((bounds) => bounds[this.maxProperty]);
+    const minimum = Math.min(...mins);
+    const maximum = Math.max(...maxes);
+
+    let distributionSpacing =
```

Comment:
Final code style suggestion: this is effectively an if/else situation, and for clarity, we should not even compute the value that you will overwrite in the isNaN if. I think it's best to write these kinds of things as a ternary expression:
```js
    const distributionSpacing = isNaN(distributeValue) ? 
      (maximum - minimum - effectiveExtent) / (boundingBoxes.length - 1) : distributeValue;
```
But also: is `isNaN` the best check? At this point I would expect `distributeValue` to be guaranteed to be undefined or a number, and not something invalid.

Reply:
Changed it. `isNaN` works well. I remember having issues with `distributeValue === undefined` while playing with the functionality of the tool. Not sure what it was about. Sorry.